cf-java-monitoring-custom-metrics-clients/cf-custom-metrics-clients-java/src/main/java/com/sap/cloud/cf/monitoring/java/CustomMetricRegistry.java

```diff
@@ -0,0 +1,63 @@
+package com.sap.cloud.cf.monitoring.java;
+
+import java.util.concurrent.TimeUnit;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.codahale.metrics.MetricRegistry;
+import com.sap.cloud.cf.monitoring.client.MonitoringClient;
+import com.sap.cloud.cf.monitoring.client.MonitoringClientBuilder;
+import com.sap.cloud.cf.monitoring.client.configuration.CFConfigurationProvider;
+import com.sap.cloud.cf.monitoring.client.configuration.ConfigurationProvider;
+import com.sap.cloud.cf.monitoring.client.configuration.CustomMetricsConfiguration;
+import com.sap.cloud.cf.monitoring.client.configuration.CustomMetricsConfigurationFactory;
+
+public class CustomMetricRegistry extends MetricRegistry {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CustomMetricRegistry.class);
+    private static volatile CustomMetricRegistry instance = null;
+
+    private ConfigurationProvider configProvider;
+    private CustomMetricsConfiguration customMetricsConfig;
+    private CustomMetricsReporter reporter;
+
+    public static MetricRegistry get() {
+        if (instance == null) {
+            synchronized (CustomMetricRegistry.class) {
+                if (instance == null) {
+                    instance = new CustomMetricRegistry();
+                }
+            }
+        }
+        return instance;
+    }
+
+    protected CustomMetricsReporter getReporter() {
+        return reporter;
+    }
+
+    private CustomMetricRegistry() {
+        configProvider = getConfigProvider();
+        customMetricsConfig = CustomMetricsConfigurationFactory.create();
+        initializeAndStartReporter();
+    }
+
+    private CFConfigurationProvider getConfigProvider() {

```

Comment:
Why should be static. Yes, it does not depend on the instance but I would not prefer to be static.

Reply:
It is called in the constructor. During that time the instance is created but not properly initialized. Hence, it cannot  safely access all attributes of the instance. To prevent accidental accesses, it is a best practice to declare such initialization methods as static.

Reply:
done